src/compiler/transformer.ts

```diff
@@ -299,6 +310,33 @@ namespace ts {
             lexicalEnvironmentSuspended = false;
         }
 
+        function hasLeadingComments(node: Node) {
+            const realNode = getParseTreeNode(node);
+            if (!realNode) {
+                return node && node.emitNode && !!node.emitNode.leadingComments;
+            }
+            const ranges = getLeadingCommentRangesOfNode(realNode, getSourceFileOfNode(realNode));
+            return ranges && ranges.length > 0;
+        }
+
+        function groupVariableDeclarationsByLeadingCommentPresence(declarations: VariableDeclaration[]): VariableDeclaration[][] {
+            return groupBy(declarations, decl => decl.emitNode && decl.emitNode.commentContext && hasLeadingComments(decl.emitNode.commentContext)
+                ? getNodeId(decl.emitNode.commentContext)
+                : -1

```

Comment:
`groupBy` uses a sparse array, but `-1` is not a valid index in a sparse array. A node will never have an id of `0`, so that should be used instead.